apps/worker/tasks/test_results_processor.py

```diff
@@ -112,6 +113,16 @@ def update_daily_totals(
         daily_totals[test_id]["skip_count"] += 1
 
 
+def handle_parsing_error(db_session: Session, upload: Upload, exc: Exception):
```

Comment:
you have defined a fn for this one block, but are creating the `UploadError` objects inline further down below.
I believe you can move this inline as well.